tests/sles4sap/crash/azure/crash_azure.pm

```diff
@@ -13,14 +13,49 @@ use version_utils 'is_sle';
 use sles4sap::azure_cli;
 use publiccloud::instance;
 use utils;
+use publiccloud::ssh_interactive;
+
+=head2 softrestart
+    ($shutdown_time, $bootup_time) = softrestart([timeout => 600] [, scan_ssh_host_key => ?]);
+Does a softrestart of the instance by running the command C<shutdown -r>.
+Return an array of two values, first one is the time till the instance isn't
+reachable anymore. The second one is the estimated bootup time.
+=cut
+sub softrestart {
+    my ($self, %args) = @_;
+    $args{timeout} //= 600;
+    $args{scan_ssh_host_key} //= 0;
+    $args{username} //= $self->username();
+    $args{instance} //= '';
+
+    my $duration;
+
+    select_host_console();
+
+    $args{instance}->ssh_assert_script_run(cmd => 'sudo /sbin/shutdown -r +1');
+    sleep 60;
+    my $start_time = time();
+
+    # wait till ssh disappear
+    my $out = $args{instance}->wait_for_ssh(timeout => $args{timeout}, wait_stop => 1, username => $args{username});
+    # ok ssh port closed
+    record_info("Shutdown failed", "WARNING: while stopping the system, ssh port still open after timeout,\nreporting: $out")
+      if (defined $out);    # not ok port still open
+
+    my $shutdown_time = time() - $start_time;
+    die("Waiting for system down failed!") unless ($shutdown_time < $args{timeout});
```

Comment:
I don't think this `die` is ever called. If the shut down took longer than `$args{timeout}`, then it would've failed around L40-L42, so for the code to reach here, `time() - $start_time` has to be smaller than `$args{timeout}`.